winbuild/Makefile.vc

```diff
@@ -21,10 +22,20 @@ CFGSET=true
 !MESSAGE                                  Libraries can be fetched at http://pecl2.php.net/downloads/php-windows-builds/
 !MESSAGE                                  Uncompress them into the deps folder.
 !MESSAGE   WITH_SSL=<dll or static>     - Enable OpenSSL support, DLL or static
+!MESSAGE      SSL_LIBS=<File1 File2>    - OpenSSL libs to use (default: libeay32.lib ssleay32.lib) (requires WITH_SSL)
+!MESSAGE      SSL_HEADER_PATH=<path>    - Path to OpenSSL include files. (requires WITH_SSL)
+!MESSAGE      SSL_CFLAGS=...            - Extra flags to supply when compiling OpenSSL modules (requires WITH_SSL)
```

Comment:
> downstream project saves inadvertent header or lib pollution. I know its unlikely but I don't want an zlib.h which is on the openSSL path to get in the way of the zlib.h I specify via ZLIB_CFLAGS="/I%ROOT_DIR%%ZLIB_DIR%"

This is fine. So far it was discussed, the base approach, among the point you've listed, is to be able to put every dependency on a separate path. These are two different approaches

- currently used, WITH_DEVEL - all the deps are in the same folder
- the PR strives, every dependency is in its separate folder

Regarding XXX_CFLAGS - it were needed, if one would have to pass additional flags, not just includes. For includes, it were enough to say - either we use XXX_INCLUDE, or if not supplied - DEVEL_INCLUDE for a particular dependency. All the other flags are handled by cURL's own makefiles. Just as an illustration - if you look at the other PR with the nghttp2 fix, an additional flag /DNGHTTP2_STATIC is needed when linking with static nghttp2. It is what we have to handle, not what should be given to outside as fe NGHTTP2_CFLAGS.

Each dep in a separate folder, as mentioned elsewhere, is also propagated by vcpkg and others now, for long is possible in PHP as well. But we'd still expect same structure in every single dependency directory. The only exception this PR adds is possibility to use different lib names. Even, we could go by a more simple approach, to say WITH_XYZ_DEVEL, to take the base path and construct both lib and include path. Anyway, being concerned that a builder creates mess in the dependencies is likely not worth it to over complicate things.

- XYZ_LIB - lib path for libxyz
- XYZ_INCLUDE - include path for libxyz
- XYZ_LIBS - lib names for libxyz

The first two could be replaced by XYZ_DEVEL, depends on what you guys say.

Thanks.